[AppBar] Hide and Elevate on Scroll (#15522)
mui-org/material-ui: docs/src/pages/demos/app-bar/app-bar.md

## Bottom App Bar

{{"demo": "pages/demos/app-bar/BottomAppBar.js", "iframe": true, "maxWidth": 500}}

## Hide App Bar on scroll

An App Bar that hides on scroll

{{"demo": "pages/demos/app-bar/HideAppBar.js", "iframe": "true", "maxWidth": 500}}

## Elevate App Bar on sroll

An App Bar that elevates on scroll

{{"demo": "pages/demos/app-bar/ElevateAppBar.js", "iframe": "true", "maxWidth": 500}}

## Elevate App Bar on container sroll

> An App Bar that elevates on scroll of a container

Author: @mbrookes  Originally I wanted to highlight the difference between using and not using setRef.  If setRef isn't used, then window is the default scroll target.  Unfortunately, I couldn't find a good way to show the default case inside of the DemoFrame without passing in the demo's window reference and using setRef.  The DemoFrame is needed with App Bar having position = 'fixed' for Slide to show correctly.

Since the only difference between the two demo's is the scroll target, which is the demo frame in the first example and the inner Container in the second, maybe we should remove the inner Container example and rely on the documentation (TBC) to explain the default functionality?  I could also add a few sentences if you feel that would be better but it might just confused people.  What do you think?

## Bottom App Bar

{{"demo": "pages/demos/app-bar/BottomAppBar.js", "iframe": true, "maxWidth": 500}}

## Scrolling

### Hide App Bar

An App Bar that hides on scroll.

{{"demo": "pages/demos/app-bar/HideAppBar.js", "iframe": "true", "maxWidth": 500}}

### Elevate App Bar

An App Bar that elevates on scroll.

{{"demo": "pages/demos/app-bar/ElevateAppBar.js", "iframe": "true", "maxWidth": 500}}

### `useScrollTrigger([options]) => trigger`

#### Arguments

1. `options` (*Object* [optional]):

>   - `options.disableHysteresis` (*Boolan* [optional]): Defaults to `false`. Disable the hysteresis. Ignore the scroll direction to determine the `trigger` value.

Reviewer: ```suggestion
  - `options.disableHysteresis` (*Boolan* [optional]): Defaults to `false`. Disable the hysteresis. Ignore the scroll direction when determining the `trigger` value.
```
?

>   - `options.target` (*Node* [optional]): Defaults to window.

Reviewer: ```suggestion
  - `options.target` (*Node* [optional]): Defaults to `window`.
```

Reviewer: Currently, some "Defaults" statements don't use backticks but they probably all should.

>   - `options.threshold` (*Number* [optional]): Defaults to `100`. Change the `trigger` value when the vertical scroll crosses this threshold.

Reviewer: ```suggestion
  - `options.threshold` (*Number* [optional]): Defaults to `100`. The `trigger` value changes when the vertical scroll crosses this threshold.
```

#### Returns

> `trigger`: Does the scroll position match the criterions?

Reviewer: ```suggestion
`trigger`: Does the scroll position match the criteria?
```
